api-module/src/main/java/com/romanticpipe/reviewcanvas/common/security/TokenProvider.java

```diff
@@ -0,0 +1,173 @@
+
+		String accessToken = Jwts.builder()
+			.setExpiration(accessTokenValidity)
+			.setSubject(auth.getName())
+			.claim(AUTHORITIES_KEY, auths)
+			.claim(AUTH_ID, admin.getAdminAuthId())
+			.claim(ADMIN_ID, admin.getId())
+			.signWith(secretKey, SignatureAlgorithm.HS512)
+			.compact();
+
+		return accessToken;
+	}
+
+	public void createRefreshToken(AdminAuth adminAuth, Long adminId) {
+		long now = (new Date()).getTime();
+		Date refreshTokenValidity = new Date(now + 1000 * this.refreshTokenValidityTime);
+		adminAuth.setRefreshToken(Jwts.builder()
+			.setExpiration(refreshTokenValidity)
+			.claim(AUTH_ID, adminAuth.getId())
+			.claim(ADMIN_ID, adminId)
+			.signWith(secretKey, SignatureAlgorithm.HS256)
+			.compact());
+	}
+
+	public AdminInterface getAdmin(Claims claims) {
+		Collection<? extends GrantedAuthority> authorities =
+			Arrays.stream(claims.get(AUTHORITIES_KEY).toString().split(","))
+				.map(SimpleGrantedAuthority::new)
+				.collect(Collectors.toList());
+		AdminInterface admin;
+		if (authorities.stream()
+			.anyMatch(authority -> authority.getAuthority().equals(Role.SUPER_ADMIN_ROLE.toString()))) {
+			admin = this.superAdminValidator.validByAuthId(Long.parseLong(claims.get(AUTH_ID).toString()));
+		} else {
+			admin = this.shopAdminValidator.validByAuthId(Long.parseLong(claims.get(AUTH_ID).toString()));
+		}
+		return admin;
+	}
+
+	public boolean isExpiredById(long authId) throws BusinessException {
+		String refreshToken = this.adminAuthValidator.findById(authId).getRefreshToken();
+		if (isExpiredToken(refreshToken)) {
+			return true;
+		}
+		return false;
+	}
+
+	public Authentication getAuthentication(Claims claims, String token) {
+		Collection<? extends GrantedAuthority> authorities =
+			Arrays.stream(claims.get(AUTHORITIES_KEY).toString().split(","))
+				.map(SimpleGrantedAuthority::new)
+				.collect(Collectors.toList());
+		AdminInterface admin;
+		if (authorities.stream()
+			.anyMatch(authority -> authority.getAuthority().equals(Role.SUPER_ADMIN_ROLE.toString()))) {
+			admin = this.superAdminValidator.validByAuthId(Long.parseLong(claims.get(AUTH_ID).toString()));
+		} else {
+			admin = this.shopAdminValidator.validByAuthId(Long.parseLong(claims.get(AUTH_ID).toString()));
+		}
+		return new UsernamePasswordAuthenticationToken(admin, token, authorities);
+	}
+
+	public UsernamePasswordAuthenticationToken configureAuthentication(AdminInterface admin,
+		List<GrantedAuthority> authorities) {
+		UsernamePasswordAuthenticationToken auth =
+			new UsernamePasswordAuthenticationToken(admin, secretKey, authorities);
+		SecurityContextHolder.getContext().setAuthentication(auth);
+		return auth;
+	}
+
+	public boolean isExpiredToken(String refreshToken) {
+		if (refreshToken.isEmpty() || refreshToken.equals(DELETED_TOKEN)) {
+			return true;
+		}
+		Claims claims = parseClaims(refreshToken);
+		if (claims.getExpiration().before(new Date())) {
+			return true;
+		}
+		return false;
+	}
```

Comment:
아래처럼 바꾸면 코드가독성이 높아질것 같네요 그리고 new Date를 그대로 넣는건 제어할 수 있는 코드일까요? 

테스트코드를 작성한다 생각해보세요 

```suggestion 
public boolean isExpiredToken(String refreshToken, Date currentDate) {
		if (refreshToken.isEmpty() || refreshToken.equals(DELETED_TOKEN)) {
			return true;
		}
		Claims claims = parseClaims(refreshToken);
		
		return claims.getExpiration().before(currentDate);
	}
```

Reply:
테스트 코드 작성시 필요한, 제어되야 할 부분이라는 게 처음 보이네요,,
전혀 고려하지 못했던 것 같습니다.
리뷰에 맞게 반영하도록 하겠습니다.